R/shinyEnrichment_utils.R

```diff
@@ -0,0 +1,142 @@
+
+#' @title Dot Plot Enrichment Visualization
+#' @description Creates interactive dot plot for enrichment results
+#' @param enrich Enrichment result object (data.frame-like)
+#' @param numSets Number of top results to show (default: 10)
+#' @param colourBy Column name for color gradient (default: "p.adjust")
+#' @param titleForPlot Plot title
+#' @param x_axis X-axis variable ("GeneRatio" or "Count")
+#' @param textSize Y-axis text size
+#' @param colour1 Low end color
+#' @param colour2 High end color
+#' @return plotly object
+#' @importFrom rlang sym .data
+#' @importFrom ggplot2 ggplot aes geom_point scale_size_continuous theme_minimal element_text labs scale_color_gradient reorder
+#' @importFrom plotly ggplotly
+#' @aurhtor Jacob Martin
+#' @export
+
+dotPlotEnrichment <- function(enrich, numSets = 10, colourBy = "p.adjust", titleForPlot = "Title", x_axis = "GeneRatio", textSize = 10, colour1 = "red", colour2 = "blue"){
+    enrich <- enrich[order(enrich$p.adjust), ]
+    enrich <- enrich[1:numSets, ]
```

Comment:
Try to avoid the `1:whatever` style, it can have unintended results. Could do `seq(numSets)`. In other instances, `seq_along` or `seq_len` may be necessary.